scanner/src/main/java/edu/ucr/cs/riple/scanner/out/MethodInfo.java

```diff
@@ -168,4 +173,21 @@ public void setParameterNames(List<? extends VariableTree> parameters) {
                 })
             .toArray(String[]::new);
   }
+
+  /**
+   * Return string value of visibility
+   *
+   * @return "public" if public, "private" if private and "package" if the method has package
+   *     visibility.
+   */
+  private String getVisibilityOfMethod() {
```

Comment:
Do we need to reason about the "default" modifier too?

Reply:
We are not reasoning about the `default` modifier, however I believe it is better to serialize full information rather than just the information we currently need to avoid further PRs :)

There are two alternatives to this approach:
1. write `public` for `public method` and `private` for `private method` and `null` for package visibility.
2. only write `true` for public methods and `false` for all others.

I prefer the approach we are already taking as it is more readable and understandable. Let me know what you think of this. Also I am happy to hear your thoughts @lazaroclapp as well :)

Reply:
I see. I prefer the current strategy over 2.

Reply:
The current approach makes sense to me, but I believe we are missing a check for `Modifier.PROTECTED`.